olp-cpp-sdk-authentication/src/TokenProviderImplPrivate.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+#include <olp/authentication/ErrorResponse.h>
+#include <olp/authentication/Settings.h>
+#include <olp/authentication/Types.h>
+#include <olp/core/client/CancellationContext.h>
+#include <olp/core/client/OauthToken.h>
+#include <olp/core/http/HttpStatusCode.h>
+
+namespace olp {
+namespace authentication {
+
+class AutoRefreshingToken;
+
+namespace internal {
+
+class TokenProviderImplPrivate {
+ public:
+  TokenProviderImplPrivate(Settings settings,
+                           std::chrono::seconds minimum_validity);
+
+  std::string operator()() const;
+
+  client::OauthTokenResponse operator()(
+      client::CancellationContext& context) const;
+
+  ErrorResponse GetErrorResponse() const;
+
+  int GetHttpStatusCode() const;
+
+  TokenResponse GetResponse(client::CancellationContext& context) const;
+
+  bool IsTokenResponseOK() const;
```

Comment:
All public members should be documented.

Reply:
done